content/en/docs/usage/kraft_advanced.md

```diff
@@ -0,0 +1,732 @@
+If you want to use an existing application listed in `kraft list`, you can use the `-t, --template` option.
+
+```console
+$ kraft init -M -t helloworld my-helloworld-app
+[...]
+[INFO    ] Initialized new unikraft application: /home/stefan/projects/unikraft/my-helloworld-app-init
+```
+
+A new directory named `my-helloworld-app/` will be created inside you working directory, and will contain the following file structure:
+
+```text
+.
+|-- CODING_STYLE.md
+|-- Config.uk
+|-- CONTRIBUTING.md
+|-- COPYING.md
+|-- kraft.yaml
+|-- main.c
+|-- MAINTAINERS.md
+|-- Makefile
+|-- Makefile.uk
+|-- monkey.h
+`-- README.md
+```
+
+You can use the generated `Makefile` to invoke the Unikraft build system.
+
+```console
+$ cat Makefile
+UK_ROOT ?= $(PWD)/../../unikraft
+UK_LIBS ?= $(PWD)/../../libs
+LIBS :=
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+
+```
+
+You can build your application on top of this basic file structure, by adding source files and modifying the `Makefile.uk` and `Config.uk` files, as presented in the [advanced usage docs](https://unikraft.org/docs/usage/advanced/).
+
+### Overview of the `kraft init` Command
+
+You can get a full list of features by running `kraft init -h` or `kraft init --help`.
+
+```text
+Usage: kraft init [OPTIONS] [NAME]
+
+  Initializes a new unikraft application.
+
+  Start here if this is your first time using (uni)kraft.
+
+Options:
+  -t, --template NAME  Use an existing application as a template.
+  -p, --plat PLAT      Target platform.
+  -m, --arch ARCH      Target architecture.
+  -w, --workdir PATH   Specify an alternative directory for the application
+		       [default is cwd].
+  -M, --with-makefile  Create a Unikraft compatible Makefile.
+  -D, --no-deps        Do not download additional dependencies for application
+		       components.
+  -d, --dump           Dump dependencies into project directory.
+  -F, --force          Overwrite any existing files.
+  -h, --help           Show this message and exit.
+```
+
+## `kraft configure`
+
+You can configure the application by using `kraft configure`.
+The configuration will be saved inside the `.config` file.
+
+When no option is specified, you will be prompted with a list from where you can choose your desired configuration.
+Use the arrows to navigate through the list:
+
+```console
+$ kraft configure
+[?] Which target would you like to configure?: helloworld_linuxu-x86_64
+> helloworld_linuxu-x86_64
+  helloworld_kvm-x86_64
+  helloworld_xen-x86_64
+  helloworld_linuxu-arm64
+  helloworld_kvm-arm64
+  helloworld_linuxu-arm
+```
+
+You can specify the platform and architecture by using the `-p, --plat` and `-m, --arch` options:
+
+```console
+$ kraft configure -p kvm -m x86_64
+[...]
+#
+# configuration written to /media/stefan/projects/unikraft/my-helloworld-app/.config
+#
+make: Leaving directory '/home/stefan/.unikraft/unikraft'
+```
+
```

Comment:
```suggestion

The comamnd above configured the application for the `kvm` configuration and `x86_64` architecture.
The configuraion is saved in the `.config` file:

""" # replace with triple backticks (I'm not doing it because of messy suggestion mode)
$ grep '^[^#].*\(PLAT\|ARCH\)' .config
CONFIG_UK_ARCH="x86_64"
CONFIG_ARCH_X86_64=y
CONFIG_MARCH_X86_64_GENERIC=y
CONFIG_PLAT_KVM=y
"""

```